### DISCUSSION TITLE
Tag scrollviewer on the side rather then the top

### DISCUSSION THREAD

**Discussion by Joly0** [1Remote/1Remote#403] [Ideas]
**Is your enhancement related to a problem? Please describe.**

I am using premotem now for about 1 month with alot of servers (i am speaking like over 100 servers) and i have added tags for every single one so i can easier find the one i am searching (as premotem, unlike mremoteng, doesnt support folders).
Though the scrollviewer on the top is a cluttered mess, if the tag is longer than a few characters. It just looks like a very long string to and its hard to find the one tag i search. 

**Describe the solution you'd like**

The # is helping, though it would be visually way easier if the scrollviewer would be on a side, to scroll through a list of tags and they are separated like they should in my opinion.

**Describe alternatives you've considered**

Alternatives i have considered would be to increase the space between each tag, though that just would make the scrollviewer wider with more unnecessary space in between, that wouldnt be needed, if the scrollviewer were on a side.
Another alternative would be, if the tags were by their background color separated from each other. I am using the dark theme and the background is grey, it could help if every other tag is light grey. That could help improve the scrollviewer visually.

**Additional context**
Add any other context or screenshots about the enhancement here.

**Comment by VShawn** [comment]
Hi, I make PRemoteM from the idea to manage my home lab with Raspberry Pi / Nas / PT server... Since I design it for MY HOME USE, I consider manage about 20 devices with PRemoteM, so I never leave a strong management module in the prototype design. 

That's why we don't have a folders support like mRemoteNG:)

Since more and more new user with large count of servers coming in. I started to think of how to deal with large count of servers. The tag system is one of the effort for this.

Now I’m working on https://github.com/1Remote/PRemoteM/issues/282, it will let us save servers to different source. (e.g. personal server save to local database, team server save to mysql database in LAN)

---

Backing to the ticket you open, IMO 'tag on the side' may not give much help since you have lots of tags.

- still need to scrolling if large count tags displayed
- 'if the tag is longer than a few characters' it will take much horizontal display space which was leaving for showing the server list. 
- https://github.com/1Remote/PRemoteM/issues/282 will take the role of folder in the future version. Maybe a treeview will replace the current list view for servers list display.

---

Here is my approch:

I have about 50 servers, every server tagged with `work` , `personal`, `RDP`,`LAN`, `WAN` etc...

Only 2 tag been shown above: `work`, `personal`

When I looking for some server, I click the `work`, then I press `Ctrl` and click the sub-tag I looking for(like the screen shot below). In this approach I can find any server by at most 3 mouse click.

![image](https://user-images.githubusercontent.com/10143738/189796655-bf6e4fdb-d436-4ad3-9bd2-8cd0ee32a1a1.png)

**Comment by Joly0** [comment]
Hm, as far as i can tell, there is a alot of free space in the server list (atleast when using the list view, rather than card view), so there should be space for a vertical tag list there, though that was just an idea from me and i definitly see your point. 

Your idea sounds interesting too, though the need to press ctrl aswell sounds like an inconvenience to me, but that might just be me. Wouldnt it be easier then to have the top tags like they are now and then a second row with all tags within that parent-tag, which i can easily click then?

This approach could improve in terms of the overall visual clarity, though i think, that shouldnt make it harder to add new servers, which it could, as i would have to first set a parent tag and than add child-tags along with it.
And additionaly, this could bring my parent-tags down to about 5-6 (or maybe down to 3-4, currently i have about 20 which already is less then i actually need....), but others might not be able to split it up so easily and might still have to deal with this. Aswell, if the server lists grows and the tags along with it, there might be the need to add more parent-tags, which then aswell adds more clutter for the parent-tag scrollviewer.

Currently i have no better idea to solve this, maybe i find one, though i wanted to share my thoughts on this.

**Comment by VShawn** [comment]
> as far as i can tell, there is a alot of free space in the server list

Not enought when app stay in a small size window, like below. I would like to leave the window in this size so that it will not block the others window. Anddd as you can see there's no much space if i give a long name to a server.

![image](https://user-images.githubusercontent.com/10143738/189864339-dbdbb12a-8cf6-406a-b2ff-60ad19908fa8.png)

I'm planning to rebuild this list after #282 is done, haven't decided how to design a now one yet.

> Wouldnt it be easier then to have the top tags like they are now and then a second row with all tags within that parent-tag.

I am afraid that we dont have a so called `parent-tag` in our system. (Mabye we will never have it, parent-tag sounds weird). 

---

I dont have a better idea either :)

Since I have no need to manage so many servers, which means I don't have enough experience on how to manage them in a smooth approach.

**Comment by majkinetor** [comment]
Pinned tag list is never ment to be huge and is meant to be dynamic - you pin tags you work with often, and when you don't need them you unpin them. You also have a full screen of tags which is basically what you want. We might redesign this on #401. Hence I will close this since its a redudant option giving next to 0 value and still require maintenance.

**Comment by Joly0** [comment]
This doesnt make a lot of sense. Even if the tag list is supposed to be used only for tags that are used often, if the tag is longer than a few characters, the next tag will start right after the first one finished, giving 0 space between two tags and making it harder to read them.
Atleast consider adding some space between the tags.
I thought making them vertical instead of horizontal wasnt that hard, thats why i suggested that, but having to read something like this:

#Admin Servers#Application Servers#Database Servers#Active Directory#Testing Servers
Is not really easy to read. As i said in my original message, the # makes it a bit easier, but still not easy (btw, my tag list is even longer than that and i have already excluded unneeded ones).

**Reply by majkinetor** [reply]
Not sure what you are talking about, here is how it looks now:

![image](https://github.com/1Remote/1Remote/assets/85767/60babaa7-fece-4e5d-a90b-982931e1360f)

Tags will be limited in length to no more then 32 chars or so.

**Reply by VShawn** [reply]
@Joly0 maybe you can draw a prototype for us to easy understand

**Reply by majkinetor** [reply]
Adding bunch of tags above doesn't influence separation, I just checked. With it you can horizontally scroll them with mouse weel.
